### PR TITLE
PMSO-170: Implement Cancel Product Order and Unit Test

### PR DESCRIPTION
**Pull request by ddnguyen99**
### Description

Add Partial_update in views to cancel Product Order 
Add Unit Test for cancelling Product Order

### Related Jira(s)

- PMSO-170

### Checklist

- [ x] I have read and followed the [PR naming convention](https://pulsar.apache.org/contribute/develop-semantic-title/)
- [x ] My code follows the code style of this project.
- [ x] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

### Tests

python manage.py test

### CODE REVIEW

**Comment by rcs-tringuyen** [comment]
@ddnguyen99 Error: https://stackoverflow.com/questions/57406706/direct-assignment-to-the-reverse-side-of-a-related-set-is-prohibited-use-addres
How to fix: Instead of trying to create Product Order using Django models and ORM, you can leverage your API to create order, and then test out the GET functionality.

**Review by rcs-tringuyen — DISMISSED** [review]

**Comment by rcs-tringuyen** [comment]
@LongNgg @phamhuutriet tui bay review gium t do t commit vao roi ko approve dc nua

File: pmsoService/products/tests/tests_product_order.py [hunk 1]
```diff
@@ -48,25 +49,6 @@ def test_get_non_existent_product_order(self):
         response = self.client.get(reverse("product_order_detail", kwargs={"id": non_existent_id}))
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
 
-    def test_create_product_order(self):
```
Comment (phamhuutriet) [inline pmsoService/products/tests/tests_product_order.py:51]: tại sao xoá test này?

File: pmsoService/products/tests/tests_product_order.py [hunk 2]
```diff
@@ -37,6 +37,7 @@ def setUpTestData(self):
 
     def test_get_product_order(self):
         self.client.force_authenticate(user=self.user)
+        # print(f"Product_oder_id: {self.product_order.id}")
```
Comment (phamhuutriet) [inline pmsoService/products/tests/tests_product_order.py:40]: xoá dòng này đi

File: pmsoService/products/tests/tests_product_order.py [hunk 3]
```diff
@@ -79,4 +61,22 @@ def test_update_product_order(self):
...
         self.client.force_authenticate(user=self.user)
         response = self.client.delete(reverse("product_order_detail", kwargs={"id": self.product_order.id}))
-        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
\ No newline at end of file
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_cancel_product_order(self):
+        self.client.force_authenticate(user=self.user)
+        response = self.client.patch(reverse("product_order_detail", kwargs={"id": self.product_order.id}),
+                                     {"status" : "Cancelled"}, format="json")
```
Comment (phamhuutriet) [inline pmsoService/products/tests/tests_product_order.py:69]: dùng enum

File: pmsoService/products/tests/tests_product_orders.py [hunk 4]
```diff
@@ -70,19 +72,38 @@ def test_create_product_order_with_single_product(self):
         product_order = ProductOrder.objects.get(id=product_order_id)
         self.assertIsNotNone(product_order)
 
-    def test_create_product_order_with_multiple_product(self):
+    def test_create_and_get_product_order_with_multiple_product(self):
```
Comment (phamhuutriet) [inline pmsoService/products/tests/tests_product_orders.py:75]: rename như này không đúng nghĩa, nó chỉ create thôi chứ có get gì đâu

File: pmsoService/products/tests/tests_product_orders.py [hunk 5]
```diff
@@ -70,19 +72,38 @@ def test_create_product_order_with_single_product(self):
...
+                                                          self.logistic.id,
+                                                          self.deliverer.id,
                                                           mock_products_generator(product_ids))
 
         response = self.client.post(
             reverse("product_order_list_create"), MOCK_PRODUCT_ORDER, format='json'
         )
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
 
+    def test_get_product_order(self):
```
Comment (phamhuutriet) [inline pmsoService/products/tests/tests_product_orders.py:91]: nên split ra 2 test 1 for single 1 for list

File: pmsoService/products/views.py [hunk 6]
```diff
@@ -19,6 +20,17 @@ class ProductOrderRetrieveUpdateDestroyView(generics.RetrieveUpdateDestroyAPIVie
     serializer_class = ProductOrderSerializer
     lookup_field = "id"
 
+    def partial_update(self, request, *args, **kwargs):
```
Comment (phamhuutriet) [inline pmsoService/products/views.py:23]: không cần thiết, partially update đã được included trong Mixin. Dùng patch thay vì put

**Review by phamhuutriet — CHANGES_REQUESTED** [review]

File: .idea/inspectionProfiles/Project_Default.xml
Comment (rcs-tringuyen) [inline .idea/inspectionProfiles/Project_Default.xml:1]: Remove .idea files please @ddnguyen99

**Review by rcs-tringuyen — APPROVED** [review]

**Review by phamhuutriet — APPROVED** [review]

**Pull request merged**